### PR TITLE
Call rescale_output_resolution before early return

### PR DESCRIPTION
**Pull request by pierotofy**
Hello :wave:

While looking through the splatfacto implementation, I noticed that in the case of an early exit in the forward pass due to a zero radii sum the camera does not get rescaled back to its original dimensions. I think this might be a mistake? I haven't found a dataset where the condition occurs, but it seems correct to make sure the camera is rescaled back before returning.

### CODE REVIEW

**Comment by chungmin99** [comment]
> in the case of an early exit in the forward pass due to a zero radii sum the camera does not get rescaled back

Thank you for catching it! This condition can actually happen quite frequently when there's no gaussians in front (e.g., due to cropping + moving the camera away from the cropbox). 

---

The fix looks great, but I'm just wondering if we can remove duplicate code, by moving the camera rescaling logic (L745, and L763) before the radii check (L738):

https://github.com/nerfstudio-project/nerfstudio/blob/c5520fb4b1ff45907f3a6a67acc76cdbcbf08ccb/nerfstudio/models/splatfacto.py#L737-L747

And we can change `rgb = background.repeat(int(camera.height.item()), int(camera.width.item()), 1)` to `rgb=background.repeat(H, W, 1)`. 

... unless there's any dependence on the `camera` object I'm missing between L750-760.

**Comment by jb-ye** [comment]
Instead of adding this line to early return, could you move the line later to an earlier position: right after project_gaussian() call. We no longer need `camera` after that.

**Comment by jb-ye** [comment]
You can test your code by enabling `--pipeline.model.num_downscales=2` because currently it is set to zero by default.

File: nerfstudio/models/splatfacto.py [hunk 1]
```diff
@@ -747,10 +747,14 @@ def get_outputs(self, camera: Cameras) -> Dict[str, Union[torch.Tensor, List]]:
             tile_bounds,
         )  # type: ignore
 
+        # rescale the camera back to original dimensions before returning
+        camera.rescale_output_resolution(camera_downscale)
```
Comment (jb-ye) [inline nerfstudio/models/splatfacto.py:751]: You need to remove the line 773-774

Comment (chungmin99) [reply nerfstudio/models/splatfacto.py:751]: Ah yes, thanks for catching -- just fixed this

**Comment by jb-ye** [comment]
lgtm

**Review by chungmin99 — APPROVED** [review]

**Pull request merged**